clients/src/test/java/org/apache/kafka/common/metrics/MetricsTest.java

```diff
@@ -829,4 +829,59 @@ private Sensor createSensor(StatType statType, int index) {
             return sensor;
         }
     }
+
+    /**
+     * This test is to verify the deprecated {@link Metric#value()} method.
+     * @deprecated This will be removed in a future major release.
+     */
+    @Test
+    public void testDeprecatedMetricValueMethod() {
```

Comment:
Rather than duplicate the whole method, could we convert the existing method into a utility and have two tests which use that? Something like:
```
public void testSimpleStats() {
    verifyStats(m -> (double) m.metricValue());
}
public void testDeprecatedMetricValueMethod() {
    verifyStats(KafkaMetric::value);
}
private void verifyStats(Function<KafkaMetric, Double> metricValueFunc) {
   // Existing testSimpleStats method modified to apply the function
}
```
When we get rid of the deprecated method, we could remove the function and use the method directly. What do you think?

Reply:
:+1: Updated the PR.